blenderproc/python/writer/GifWriterUtility.py

```diff
@@ -0,0 +1,210 @@
+                            in milliseconds.
+    :param time_direction: If this is True, the order of the frames
+                            will be reversed.
+    """
+
+    # Generates subdirectory for .gif files
+    output_dir_path = GifWriterUtility._provide_directory(output_dir_path)
+    
+    # From WriterUtility.py
+    amount_of_frames = 0
+    for data_block in output_data_dict.values():
+        if isinstance(data_block, list):
+            amount_of_frames = max([amount_of_frames, len(data_block)])
+
+    # From WriterUtility.py
+    if amount_of_frames != bpy.context.scene.frame_end - bpy.context.scene.frame_start:
+        raise Exception("The amount of images stored in the output_data_dict"
+                        "does not correspond with the amount"
+                        "of images specified by frame_start to frame_end.")
+
+    # Sorts out keys which are just metadata and not plottable
+    keys_to_use = GifWriterUtility._select_keys(output_data_dict)
+
+    # Build temporary folders with .png collections
+    to_animate = GifWriterUtility._cache_png(keys_to_use, output_data_dict)
+
+    # Write the cache .png files to .gif files and delete cache
+    GifWriterUtility._write_to_gif(to_animate, output_dir_path, 
+                                    append_to_existing_output, 
+                                    time_constant, 
+                                    time_direction)
+
+
+class GifWriterUtility:
+    
+    @staticmethod
+    def _provide_directory(output_dir_path: str) -> str:
+        """ Generates subdirectory for .gif files if not existent """
+        output_dir_path = os.path.join(output_dir_path, 'gif_animations')
+        if not os.path.exists(output_dir_path):
+            print('\n Generate output folder: ' + output_dir_path)
+            os.makedirs(output_dir_path)
+        return output_dir_path
+
+    @staticmethod
+    def _select_keys(
+            output_data_dict: Dict[str, List[Union[np.ndarray, list, dict]]]
+            ) -> List[str]:
+        """ Sorts out keys which are just metadata and not plottable """
+        keys_to_use = []
+        for key in output_data_dict.keys():
+            value = output_data_dict[key][0]
+            value = np.array(value)
+            if np.issubdtype(value.dtype, np.string_) or len(value.shape) == 1:
+                pass # metadata
+            else:
+                keys_to_use.append(key)
+        return keys_to_use
+
+    @staticmethod
+    def _cache_png(keys_to_use: List[str], 
+            output_data_dict: Dict[str, List[Union[np.ndarray, list, dict]]]
+            ) -> Dict[str, str]:
+        """
+        Builds temporary folders with .png collections 
+        and returns the locations as dictionary. 
+        """
+
+        if not set(keys_to_use) <= set(output_data_dict.keys()):
+            raise ValueError("The keys_to_use list must be contained in"
+                            "the list of keys from the output_data_dict!")
```

Comment:
The error message contains more info than actually is being checked. There is no check if there elements which are not in the output keys. 

```suggestion
            raise ValueError("There are more requested keys than are available in the output data.")
```

Reply:
It's actually true what I wrote. It checks if it is a subset. So it's like:
```python
set([1,4]) <= set([1,2,3])
```
```
>>> False
```
It does not behave like a `len(list)` comparison.

Reply:
Interesting I haven't seen that before.